applications/ecallmgr/src/ecallmgr_fs_recordings.erl

```diff
@@ -1,228 +1,45 @@
-                'true' -> props:set_value(<<"Publish-Channel-State">>, 'false', Props);
-                _ -> Props
-            end;
-        _Value -> Props
-    end.
-
--spec handle_record_event(atom(), kz_proplist(), kz_proplist()) -> 'ok'.
-handle_record_event(Node, [UUID | FSProps], Options) ->
-    kz_util:put_callid(UUID),
-    Props = init_props(FSProps, Options),
-    process_event(UUID, Props, Node),
-    maybe_publish_record_event(Props).
-
--spec process_event(api_binary(), kz_proplist(), atom()) -> any().
-process_event(UUID, Props, Node) ->
+-spec handle_record_stop(map()) -> any().
+handle_record_stop(#{node := Node, call_id := UUID, payload := JObj}) ->
     kz_util:put_callid(UUID),
-    EventName = kzd_freeswitch:event_name(Props),
-    process_specific_event(EventName, UUID, Props, Node).
-
--spec process_specific_event(ne_binary(), api_binary(), kz_proplist(), atom()) -> any().
-process_specific_event(<<"RECORD_STOP">>, UUID, Props, Node) ->
-    IsLocal = props:is_true(<<"Force-Publish-Event-State">>, Props, 'true')
-        orelse (props:is_true(<<"Publish-Channel-State">>, Props, 'true')
-                andalso handling_locally(Props)
-               ),
-    Args = [IsLocal, kzd_freeswitch:media_recorder(Props), Props, UUID, Node],
-    kz_util:spawn(fun maybe_store_recording/5, Args);
-process_specific_event(<<"RECORD_START">>, _UUID, _Props, _Node) -> 'ok';
-process_specific_event(_Event, _UUID, _Props, _Node) ->
-    lager:debug("event ~s for callid ~s not handled in recordings (~s)", [_Event, _UUID, _Node]).
-
-
--spec maybe_publish_record_event(kz_proplist()) -> 'ok'.
-maybe_publish_record_event(Props) ->
-    case props:is_true(<<"Force-Publish-Event-State">>, Props, 'true')
-        orelse (props:is_true(<<"Publish-Channel-State">>, Props, 'true')
-                andalso handling_locally(Props)
-               )
-    of
-        'true' -> ecallmgr_call_events:process_channel_event(Props);
-        'false' -> lager:debug("not publishing record event ~s", [kzd_freeswitch:event_name(Props)])
-    end.
+    IsLocal = handling_locally(JObj),
+    MediaRecorder = kz_call_event:custom_channel_var(JObj, <<"Media-Recorder">>),
+    maybe_store_recording(IsLocal, MediaRecorder, JObj, UUID, Node).
 
 -spec maybe_store_recording(boolean(), api_binary(), kz_proplist(), ne_binary(), atom()) ->
                                    'ok' |
                                    'error' |
                                    ecallmgr_util:send_cmd_ret() |
                                    [ecallmgr_util:send_cmd_ret(),...].
-maybe_store_recording('false', _, _Props, _CallId, _Node) -> 'ok';
+maybe_store_recording('false', _, _JObj, _CallId, _Node) -> 'ok';
 maybe_store_recording('true', <<"kz_media_recording">>, _Props, _CallId, _Node) -> 'ok';
-maybe_store_recording('true', _, Props, CallId, Node) ->
-    case kzd_freeswitch:ccv(Props, <<"Media-Transfer-Destination">>) of
+maybe_store_recording('true', _, JObj, CallId, Node) ->
```

Comment:
You call it `JObj` here but then create `JObj` in one of the clauses. This is bound to crash as the passed in `JObj` won't match the `kz_json:from_list(...)` construction.